### PR TITLE
Add command to not close tabs in sidebar

### PR DESCRIPTION
**Pull request by jo16oh**
I made this request because of the reasons listed below. I am new to programming, so sorry if I have made any mistakes.

### 1. Prevent closing plugin tabs in the sidebar

I open many plugins in the sidebar, and for such use cases, closing tabs by `ctrl + W` in the sidebar is very problematic because tabs opened by plugins can't be pinned. The context menu is enough to close tabs in the sidebar because I use the sidebar as a stable workspace.

### 2. To open Daily Notes in the sidebar

I tried to open Daily Notes in the sidebar, but it wasn't comfortable because it closes when I press `ctrl + W` by mistake. If it's pinned, it won't switch to the next day's Daily Notes automatically.

## Changes

- Added command "Close current tab (if not in the sidebar)"

### CODE REVIEW

**Comment by hdykokd** [comment]
Thank you for your contribution!

Does this mean you want to create a command like "close unpinned tabs in the main area"?
I have looked at the implementation, but determining whether it is in the sidebar based on the contents of the DOM elements seems to be prone to bugs, so it would be better to use the Obsidian API.

How about the following code?

```ts
const isTabGroupInMainArea = app.workspace.rootSplit.children.find(c => {
  return c.id === app.workspace.activeTabGroup.id
})
```

It is not explicitly stated, but looking at the HTML of rootSplit, it seems safe to assume that this refers to the main area.

- https://docs.obsidian.md/Reference/TypeScript+API/Workspace/rootSplit
- https://github.com/obsidianmd/obsidian-api/blob/2ccfad1a10c8da9066bf2c502f801b1a237f4ce4/obsidian.d.ts#L4729-L4738

File: src/main.ts [hunk 1]
```diff
@@ -24,7 +24,33 @@ export default class AdvancedCloseTab extends Plugin {
         this.detachLeafIfUnpinned(leaf);
       },
     });
+    
+    this.addCommand({
+      id: 'advanced-close-tab-close-current-tab-if-not-in-sidebar',
```
Comment (hdykokd) [inline src/main.ts]: It seems that it is not necessary to include `advanced-close-tab-` here, as Obsidian automatically prefixes the Plugin ID (I didn't know this when I created this plugin, so the other commands are not like that...).

https://docs.obsidian.md/Plugins/Releasing/Submission+requirements+for+plugins#Don't+include+the+plugin+ID+in+the+command+ID

So, could you please change it to `close-current-tab-if-not-in-sidebar`?

**Comment by hdykokd** [comment]
By the way, how about changing it to `Close current tab if in main area?`

**Comment by jo16oh** [comment]
I really appreciate your detailed suggestion! I've modified the code based on your advice, but there's a small change from your suggestion.

In the case of detecting the tab's placement using the rootSplit content, it won't close the tab in new windows (which I consider as the "main area"). Therefore, I've updated the code to determine the tab's placement based on the left/rightSplit content instead.

If I had any mistakes or unnatural implementation, please let me know. Thank you.

File: src/main.ts [hunk 2]
```diff
@@ -7,7 +7,7 @@ export default class AdvancedCloseTab extends Plugin {
 
   addCommands() {
     this.addCommand({
-      id: 'advanced-close-tab-close-current-tab',
+      id: 'close-current-tab',
```
Comment (hdykokd) [inline src/main.ts]: This would become a breaking change, so it would be preferable to make the change in a separate pull request.
For instance, re-registration of Hot Keys will be necessary. That was the reason I was avoiding changing the existing command IDs. Sorry for the lack of explanation.
Since this plugin is not registered as a community plugin, I think the users will understand the changes...

File: src/main.ts [hunk 3]
```diff
@@ -24,9 +24,41 @@ export default class AdvancedCloseTab extends Plugin {
...
 
+        const leaf = this.app.workspace.getLeafById(activeLeaf.id);
+        if (!leaf) return;
+
+        const isTabGroupInRightSidebar = app.workspace.rightSplit.children.find(c => {
+          return c.id === app.workspace.activeTabGroup.id
+        });
+        const isTabGroupInLeftSidebar = app.workspace.leftSplit.children.find(c => {
+          return c.id === app.workspace.activeTabGroup.id
+        });
```
Comment (hdykokd) [inline src/main.ts:48]: 👍

File: src/main.ts (hunk 2 above)
Comment (hdykokd) [reply src/main.ts]: I'm sorry for the trouble, but could you possibly revert the changes here?

Comment (jo16oh) [reply src/main.ts]: Thank you for the review. I did not think enough about the impact it would have. I've reverted the changes.

For the new command ID, can I delete the plugin name prefix? Or should I left the prefix to keep consistency?

Comment (jo16oh) [reply src/main.ts]: ✗ left → ○ leave 😓

Comment (hdykokd) [reply src/main.ts]: > For the new command ID, can I delete the plugin name prefix?

Yes, for the new command, we would like to go without adding the prefix.

Comment (jo16oh) [reply src/main.ts]: I understand. I've modified it 🤲

**Comment by hdykokd** [comment]
@hajime-nagisa Thank you for contributing! I've released it as [2.1.1](https://github.com/hdykokd/obsidian-advanced-close-tab/releases/tag/2.1.1)
